be1-go/message/answer/error.go

```diff
@@ -13,6 +22,13 @@ func (e *Error) Error() string {
 	return e.Description
 }
 
+func (e *Error) Wrap(description string) *Error {
+	return &Error{
+		Code:        e.Code,
+		Description: fmt.Sprintf(description+": %v", e.Description),
+	}
+}
```

Comment:
From a design perspective, I see that almost all of the application uses this `Wrap()` and effectively returns an `answer.Error`. While it's good that you've a centralized error management system, this also means that all the application is aware of a JSON-RPC error structure.

If you think about it, that's weird: why should some part of the code dealing with, say, application-level logic have to know about JSON-RPC and its `answer.Error` format? If in the future we were to replace WebSocket+JSON-RPC with something else that requires a different format... you wouldn't want other parts of the codebase to be aware of the change. 

So, how do we go from here to there ?
I would strongly suggest separating the error itself (which doesn't have to be in the message/answer package) from the _rendering_ of the error in the protocol. In the "Clean Architecture" style (or in the "Hexagonal Architecture", if you prefer that kind of thing) you would have a "Presenter" layer taking the error in and producing a protocol-compatible output (hexagonal arch. people call it an "adapter").
In practice, the code is simple enough that this is not a major change, but it's an important one. One simple way to put this in practice would be having an Error structure that's internal to the app (possibly without JSON encoding annotations), and the JSON-RPC -level Error structure (with JSON encoding annotations) that could be created from the internal one. 

To clarify: you don't need to do this in this PR, but I hope you understand the architectural implications :)

Reply:
Yes we already planned to change that. We think about using **github.com/pkg/errors** to handle our errors as it can be combine with zerolog to be able to get an error logging with Stacktrace. Therefore also solving the problem of the ugly wrap we created :)